proposals/11551-kubernetes-native-api/TestPlan.md

```diff
@@ -0,0 +1,87 @@
+| (KFP DB mode) Create pipeline and a pipeline version and re-create the pipeline & version using CRs in K8s mode                                                                         | Match the pipeline and pipeline version (DB vs K8s)     | Pipeline and pipeline version should match exactly except for ID?                                                                                                   |
+| (KFP DB mode) Create pipeline and 2 pipeline versions with same spec and and re-create the pipeline & version using CRs (in K8s mode)                                                   | Match the pipeline and pipeline versions (DB vs K8s)    | Pipeline and pipeline version should match exactly except for ID?                                                                                                   |
+| (KFP DB mode) Create pipeline and 2 pipeline versions with different specs and and re-create the pipeline & version using CRs (in K8s mode)                                             | Match the pipelines and pipeline versions (DB vs K8s)   | Pipeline and pipeline version should match exactly except for ID?                                                                                                   |
+| (KFP DB mode) Create pipeline and a pipeline version and and re-create the pipeline & version using CRs but with different **pipelineName** than the one in pipelineSpecs (in K8s mode) |                                                         | Should this be allowed? Will webhooks throw a validation error                                                                                                      |
+| (K8s mode) Create pipeline run for a newly migrated pipeline associated with the default experiment                                                                                     | Get Run details and validate                            | Pipeline run should be allowed and the experiment association (to the default experiment) should stay the same                                                      |
+| (K8s mode) Create an experiment and a pipeline run associating it with a migrated pipeline                                                                                              | Get Experiment and Run details and validate             | Pipeline run should be allowed and the experiment association (to the specific experiment) should be correctly saved in the DB                                      |
+| (K8s mode) Rerun a pipeline run associated with an existing experiment and a migrated pipeline version                                                                                  | Get Experiment and Run details and validate             | Pipeline run should be allowed and the experiment association (to the specific experiment) should stay inplace, but how does it remain the same if ID is different? |
```

Comment:
Experiments aren't impacted by this feature so if you can reduce some of these test cases with that in mind, please do.